File: exercises/python_playwright/Non-Breaking_Space/non_breaking_space_playwright.py
from playwright.sync_api import Page
import re


class NBSPPage:
    """Page object for the non-breaking space exercise."""

    def __init__(self, page: Page):
        """Initialize the page and target button locator."""
        self.page = page
        self.button = page.get_by_role("button", name=re.compile(r"My\s*Button"))

    def open(self) -> None:
        """Open the exercise page."""
        self.page.goto("http://uitestingplayground.com/nbsp")

    def click_button(self) -> None:
        """Click the target button."""
        self.button.click()

File: exercises/javascript_playwright/non-breaking-space/nonBreakingSpacePlaywright.js
export class NbspPage {
    /**
     *
     * @param {import('@playwright/test').Page} page
     */
    constructor(page) {
        this.page = page;
        this.button = page.getByRole("button", { name: /^My\s+Button$/ });
    }

    async openExercisePage() {
        await this.page.goto("http://uitestingplayground.com/nbsp");
    }

    async clickButton() {
        await this.button.click();
    }
}
